### DISCUSSION TITLE
eglot + typescript shows shortened eldoc content for the same symbol based on position

### DISCUSSION THREAD

**Discussion by gdanov** [joaotavora/eglot#1354] [Q&A]
when pointed at symbol that's "alone" eldoc shows the full jsdoc. When pointed at the same symbol, but inside function call (for example) the shortened version is shown. I have configured `(setq eldoc-echo-area-use-multiline-p 10)` just to be extra-safe but this seems to change nothing. I can't figure out how to debug the eglot communication and am not sure if the problem is caused by eglot or eldoc

Here are screenshots
<img width="1728" alt="Screenshot 2024-01-26 at 11 10 40" src="https://github.com/joaotavora/eglot/assets/957113/f8d8e2b5-6734-4117-9a80-01b1a0bb30f5">
<img width="1728" alt="Screenshot 2024-01-26 at 11 12 11" src="https://github.com/joaotavora/eglot/assets/957113/1a6b695f-0f90-4bc3-bc98-ed0d39edd43f">

**Comment by joaotavora** [comment]
> I can't figure out how to debug the eglot communication 

You should be able to type `M-x eglot-events-buffer` and land on a buffer that has all the transcript for your session.

> and am not sure if the problem is caused by eglot or eldoc

The "can't debug communication" problem or the "shortened doc" problem?

To the shortened doc problem, I think it's not caused by either Eglot or Eldoc.  I think it's just your server serving you that information which it thinks is best.  Eglot always just asks "any docs here?" and sometimes server return more and sometimes they return less.

**Reply by gdanov** [reply]
Thanks. Reproduced exact same behavior in VS Code :( looks like langserver "feature"
